app/src/ui/history/commit-summary.tsx

```diff
@@ -15,14 +15,95 @@ interface ICommitSummaryProps {
   readonly files: ReadonlyArray<FileChange>
   readonly emoji: Map<string, string>
   readonly gitHubUser: IGitHubUser | null
+
+  /**
+   * Whether or not the commit body container should
+   * be rendered expanded or not. In expanded mode the
+   * commit body container takes over the diff view
+   * allowing for full height, scrollable reading of
+   * the commit message body.
+   */
   readonly isExpanded: boolean
+
   readonly onExpandChanged: (isExpanded: boolean) => void
 }
 
 interface ICommitSummaryState {
+  /**
+   * The commit message summary, i.e. the first line
+   * in the commit message. Note that this
+   * may differ from the body property in the commit object
+   * passed through props, see the createState method for more
+   * details.
+   */
+  readonly summary: string
+
+  /**
+   * The commit message body, i.e. anything after the
+   * first line of text in the commit message. Note that this
+   * may differ from the body property in the commit object
+   * passed through props, see the createState method for more
+   * details.
+   */
+  readonly body: string
+
+  /**
+   * Whether or not the commit body text overflows
+   * its container. Used in conjunction with the isExpanded
+   * prop
+   */
   readonly isOverflowed: boolean
 }
 
+const maxSummaryLength = 72
+
+/**
+ * Removes whitespace characters from the end of the
+ * string
+ */
+function trimTrailingWhitespace(value: string) {
+  return value.replace(/\s+$/, '')
```

Comment:
I think you can just return the result of `\S.*\S`. This will give you back the string with leading **and** trailing whitespace removed.

Reply:
For that behavior, why not use `value.trim()`?

Reply:
I _think_ we only care about trailing white space here. In fact, we probably don't even care about that for the body. This is just a way to make sure we don't wrap and add a newline should the summary end with a bunch of white space

Reply:
There’s [`String#trimRight()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trimRight) for that usecase.

Reply:
<img width="688" alt="screen shot 2017-09-08 at 10 45 40 am" src="https://user-images.githubusercontent.com/13760/30217028-e5adbf48-9482-11e7-8e52-49ed55be3392.png">

Makes me think we shouldn't use that.